include/BSMPT/minimum_tracer/minimum_tracer.h

```diff
@@ -0,0 +1,830 @@
+   * @param output if true tracking output is printed on the screen
+   * @param unprotected if true we dont check the hessian
+   */
+  std::vector<Minimum> trackPhase(double &globMinEndT,
+                                  std::vector<double> point,
+                                  double currentT,
+                                  double finalT,
+                                  double dT        = 3,
+                                  bool output      = true,
+                                  bool unprotected = false);
+
+  /**
+   * @brief trackPhase
+   * @param point start point for tracking
+   * @param currentT start point temperature for phase tracking
+   * @param finalT end point temperature
+   * @param dT initial temperature step size
+   * @param output if true tracking output is printed on the screen
+   * @param unprotected if true we dont check the hessian
+   */
+  std::vector<Minimum> trackPhase(std::vector<double> point,
+                                  double currentT,
+                                  double finalT,
+                                  double dT        = 3,
+                                  bool output      = true,
+                                  bool unprotected = false);
+
+  /**
+   * @brief Calculates the VEV splitting when Hessian matrix gets a single
+   * negative eigenvalue.
+   *
+   * @param point where grad is zero and hessian is not positive definite
+   * @param T is the temperature.
+   */
+  void CalculateVEVSplittings(std::vector<double> point, double T);
+
+  /**
+   * @brief bool to store whether flat directions are found
+   */
+  bool flat_dirs_found = false;
+
+  /**
+   * @brief storage of all non-flat VEV-directions
+   */
+  std::vector<int> NonFlatDirections;
+
+  /**
+   * @brief storage of indices of flat 1D directions in VEV basis
+   */
+  std::vector<std::size_t> flat_1D_dirs;
+
+  /**
+   * @brief storage of indices of flat 2D directions in VEV basis
+   */
+  std::vector<std::vector<std::size_t>> flat_2D_dirs;
+
+  /**
+   * @brief storage of indices of flat 3D directions in VEV basis
+   */
+  std::vector<std::vector<std::size_t>> flat_3D_dirs;
+
+  /**
+   * @brief List of group elements allowed by the potential
+   *
+   */
+  std::vector<Eigen::MatrixXd> GroupElements;
+
+  /**
+   * @brief Reduce the VEV into the same principal quadrant.
+   *
+   * @param point to be rotated
+   */
+  void ReduceVEV(std::vector<double> &vev);
+
+  /**
+   * @brief Reduce the VEV of the minimum into the same principal quadrant.
+   *
+   * @param min to be rotated
+   */
+  void ReduceVEV(Minimum &min);
+
+  /**
+   * @brief WarpPath
+   * @param path
+   * @param T1
+   * @param F1
+   * @param T2
+   * @param F2
+   * @return wraped path
+   */
+  std::vector<std::vector<double>>
+  WarpPath(std::vector<std::vector<double>> path,
+           std::vector<double> T1,
+           std::vector<double> F1,
+           std::vector<double> T2,
+           std::vector<double> F2);
```

Comment:
Here you have a lot pass by value for vectors which lead to a lot of copies
```suggestion
  WarpPath(const std::vector<std::vector<double>>& path,
           const std::vector<double>& T1,
           const std::vector<double>& F1,
           const std::vector<double>& T2,
           const std::vector<double>& F2);
```

Reply:
I've implemented this change.